### ISSUE TITLE
FreqUI Telegram WebApp

### ISSUE THREAD

**Issue by xsa-dev** [freqtrade/frequi#1807]
**Is your feature request related to a problem? Please describe.**
Initially (as I remember) Freqtrade operated through Telegram because it's very convenient, and now, but FreqUI really makes working with the terminal even more convenient and the trading results become better.

**Describe the solution you'd like**
Not so long ago, web-app applications for Telegram appeared, there are libraries that add webapp capabilities to the application and it's not very difficult.

**Describe alternatives you've considered**
It would be convenient if support for a webapp through a bot button or a button in the Telegram menu appeared.

**Additional context**
I can make a pull request with the specified modification if required. =)

**Comment by xmatthias** [comment]
A quick research shows that all telegram webapps do is open a web-page (maybe there's more to it that i didn't find quickly?) - which would mean that the design would include exposing freqUI and the bot API directly to the internet, which is something we neither support, and strongly discourage (we're aware that we cannot fully prevent this - though i'd not consider this safe).

The only way i see this as something we would / should support is if the path of communication works as follows:
User -> Telegram servers -> Telegram bot backend (the python process)

This last part is then responsible to 
* serve the frontend
* serve the backend


If this channel thorugh telegram servers is not possible - then this is not something we'll be supporting, as it would entail exposing the API directly to the internet.

It's however pretty simple to get freqUI securely to your device by using a VPN / wireguard network (e.g. tailscale, which is easy to setup) - which you'll install on both the server, and on your phone - and have a secure tunnel to access freqUI.

---
Maybe you can describe what this would entail in your eyes (also / mainly from a security perspective)? If the above points are wrong / not how telegram webapps work, i'd like to get links to these features.

without that - i'll treat this as a question - which has been answered by "use a VPN / tailscale" above.

**Comment by xsa-dev** [comment]
Not exactly, the open pages are made for open bots.

At the moment, I suggest using it only within your own network (Virtual Private Network) - scripts for adding display capabilities in the Telegram browser do not require a constant internet connection, but will require an SSL certificate (a self-signed one is also fine), which in turn will indirectly improve the security of FreqUI (just a little bit).

I have set up FreqUI in my internal network which is accessible to me both from a mobile device and a computer. I code on my server from an iPhone and see that it is already more or less adapted for mobile devices and most likely will be able to display in Telegram as well.

In my understanding, people who will use the UI outside their internal network will do so at their own risk, and Freqtrade cannot provide any security guarantees in such a case.

Yes, from my point of view, I have more of a question than a task - will my pull request be considered if I make such an enhancement for the public myself, and will it be considered an improvement?

**Comment by xmatthias** [comment]
Enhancements that provide improved mobile views - sure.

If it's about telegram - it'll depend on the change itself (what it would entail)
Working from the assumption that we do not want users to expose their api to the internet - it'll only work within your private network.

What exactly are the benefits (when using this through telegram) that we'd expect over simply opening your browser - considering that you're already in a private network?

**Comment by xsa-dev** [comment]
The key advantage is the convenience that allows you to open the web interface and view charts on tablets, phones, and desktops with the press of a button in a Telegram bot, even in the absence of a browser. All within a trusted network. While it's essentially the same as opening a browser, it eliminates the dependency on a browser and vice versa. If a bot and UI are set up, and there are several of these bots, it's more convenient for the owner to gather them in one place and monitor them.

Freqtrade has long been using python-telegram-bot, so it's feasible to integrate Telegram JS for the webapp as well. This enhancement not only streamlines the user experience but also centralizes control, making it an indispensable tool for those looking to efficiently manage their trading strategies without the hassle of juggling multiple platforms or devices. Embrace the future of trading with our cutting-edge solution that brings unparalleled ease and functionality right at your fingertips.

**Comment by xmatthias** [comment]
I'm sorry, i don't see that. You can put a link to your desktop and use it like any other app.

From the telegram docs, it seems like the UI must import / load a telegram js.
The UI will **not** load a javascript file from a random telegram server. Any PR trying to add such a thing will be rejected.

Aside from that - it's really unknown if it breaks regular usage in web-apps (the telegram docs don't seem to mention that)- and what other side-effects this will cause.

Can provide me with documentation that confirms that this will not cause any problems to regular browser usage?

**Comment by xsa-dev** [comment]
Yeah, sometimes I change the emphasis in AI to better convey the meanings of my words, please take it as a joke)

So, I couldn't manage to save telegram.js locally and load it into the page to be able to send commands back to the bot through Telegram - for that, you need to connect to Telegram via https://telegram.org/js/telegram-web-app.js. The results of the experiment are in the branch of the fork [python-telegram-bot](https://github.com/xsa-dev/python-telegram-bot/tree/tg-local-telgram_js).

However, it turns out, for example, to display any page from a button sent by the bot in the web application, look at how the page from google.com looks on a mobile device through telegram-webapp, I think such a button could be added without providing API access. Because Telegram itself doesn't work without the internet.

<img src="https://github.com/freqtrade/frequi/assets/16959353/9d42472b-87b3-49b9-9850-163a4a82c583" height="400">
<img src="https://github.com/freqtrade/frequi/assets/16959353/f7f6a597-83c5-4cc8-a7a4-e2715f214711"
 height="400">

**Comment by xmatthias** [comment]
After having seen the UI ... This is not something that i think makes sense.
It's not full-screen - and hence offers degraded usability over a native Tab.

Also the requirement to load a javascript file from a random remote server is not something i deem acceptable.

As mentioned above - happy to get PR's for usability improvements on mobile - or also for improved documentation (for example, how to setup a private network on your phone - though I'd always try to point to upstream documentation in most cases - as we don't want to expand the scope of what we try to support and keep updated).

This particular case however is not something we'll be considering.
As such - i'll close this as "not planned" - though the original question (get freqUI on phone) can easily be accomplished
